cmd/keytransparency-server/main.go

```diff
@@ -121,11 +123,17 @@ func main() {
 	// Create gRPC server.
 	queue := mutator.MutationQueue(mutations)
 	ksvr := keyserver.New(tlog, tmap, logAdmin, mapAdmin,
-		entry.New(), auth, authz, domains, queue, mutations)
+		entry.New(), authz, domains, queue, mutations)
 	grpcServer := grpc.NewServer(
 		grpc.Creds(creds),
+		// All streaming methods are unauthenticated for now.
 		grpc.StreamInterceptor(grpc_prometheus.StreamServerInterceptor),
-		grpc.UnaryInterceptor(grpc_prometheus.UnaryServerInterceptor),
+		grpc.UnaryInterceptor(grpc_middleware.ChainUnaryServer(
+			grpc_prometheus.UnaryServerInterceptor,
+			authentication.UnaryServerInterceptor(map[string]grpc_auth.AuthFunc{
+				"/google.keytransparency.v1.KeyTransparency/UpdateEntry": authFunc,
```

Comment:
Can we avoid hardcoding this? Presumably there will be a bunch of methods under the same prefix x/y in future.

Reply:
We could replace this with a regex in the future if we need to, but simpler seems easier to test and maintain for the moment.